### PR TITLE
NancyProvider with custom bootstrapper

### PR DESCRIPTION
**Pull request by mrstebo**
I've modified the existing NancyProvider class so that we can inject a custom DefaultNancyBootstrapper.

### CODE REVIEW

**Comment by luisrudge** [comment]
@mrstebo your PR is using tabs instead of spaces

**Comment by mrstebo** [comment]
@luisrudge Ah, my bad! I'll swap them out.

File: Microphone.Nancy/NancyProvider.cs [hunk 1]
```diff
@@ -6,14 +6,13 @@
 
 namespace Microphone.Nancy
 {
-    public class NancyProvider : IFrameworkProvider
+    public class NancyProvider<TBootstrapper> : IFrameworkProvider where TBootstrapper : DefaultNancyBootstrapper
```
Comment (rogeralsing) [inline Microphone.Nancy/NancyProvider.cs]: I think `: DefaultNancyBootstrapper` should be `: INancyBootstrapper` instead as that is what is assigned in the `new NancyHost(...`

Comment (mrstebo) [inline Microphone.Nancy/NancyProvider.cs]: Sounds good. I guess that adds a bit more flexibility if someone doesn't want to just inherit from the default bootstrapper.

**Pull request merged**